include/layers/Layer.hpp

```diff
@@ -1,5 +1,11 @@
 #pragma once
+#include <omp.h>
+
```

Comment:
Avoid re-evaluating “Auto” logic every call. Resolve once (feature flags + environment + problem size) and cache in the layer/context.


```
// Called once per layer or first use
inline ParBackend resolve_auto_once(const ParOptions& opt, std::size_t n) noexcept {
#if defined(HAS_OMP)
  if (n >= opt.min_parallel_n) return ParBackend::OMP;
#elif defined(HAS_TBB)
  if (n >= opt.min_parallel_n) return ParBackend::TBB;
#elif defined(HAS_THREADS)
  if (n >= opt.min_parallel_n) return ParBackend::Threads;
#endif
  return ParBackend::Seq;
}

inline ParBackend select_backend(const ParOptions& opt, std::size_t n) noexcept {
  if (opt.backend != ParBackend::Auto) return opt.backend;
  static ParBackend cached = resolve_auto_once(opt, n); // or store in the layer
  return cached;
}
```